test_util/testutil.cc

```diff
@@ -191,7 +193,7 @@ std::string RandomName(Random* rnd, const size_t len) {
 
 CompressionType RandomCompressionType(Random* rnd) {
   auto ret = static_cast<CompressionType>(rnd->Uniform(6));
```

Comment:
Not that this is your bug, but 6 seems wrong here.  The builtin types go from 0x0 to 0x7 (so 8 of them)